pgmpy/factors/Factor.py

```diff
@@ -76,7 +76,11 @@ def __init__(self, variables, cardinality, values):
         >>> phi
         <Factor representing phi(x1:2, x2:2, x3:2) at 0x7f8188fcaa90>
         """
+
         values = np.array(values)
+        
+        if values.dtype != int and values.dtype != float:
```

Comment:
@yashu-seth Sorry, my bad here. You are correct. I think I assumed not equal as equal to.